### ISSUE TITLE
Tracker for `scatter_reduce` additional reduction options

### ISSUE THREAD

**Issue by mikaylagawarecki** [pytorch/pytorch#94083]
### 🚀 The feature, motivation and pitch

- [ ] logsumexp #31394
- [ ] indices of max and min #80439 #83980

**Comment by alexshtf** [comment]
The LogSumExp() reduction appears extremely useful in learning to rank scenarios, where we want to do a  list-wise cross-entropy loss on mini-batches, where we know the query each sample belongs to.
